### PR TITLE
docs: Update Consul K8s compatibility matrix

### PR DESCRIPTION
**Pull request by david-yu**
Add 1.12.0 notes and also add VMware Tanzu compatibility notes.

### CODE REVIEW

File: website/content/docs/k8s/installation/compatibility.mdx [hunk 1]
```diff
@@ -42,6 +43,13 @@ Consul Kubernetes delivered Red Hat OpenShift support starting with Consul Helm
 - Red Hat OpenShift is only supported for OpenShift 4.4.x and above.
 - Only the default CNI Plugin, [OpenShift SDN CNI Plugin](https://docs.openshift.com/container-platform/4.9/networking/openshift_sdn/about-openshift-sdn.html) is currently supported.
 
+## VMware Tanzu Kubernetes Grid and Tanzu Kubernetes Grid Integrated Edition
+
+Consul Kubernetes is [certified](https://marketplace.cloud.vmware.com/services/details/hashicorp-consul-1?slug=true) for both VMware Tanzu Kubernetes Grid and VMware Tanzu Kubernetes Integrated Edition.
```
Comment (karl-cardenas-coding) [inline website/content/docs/k8s/installation/compatibility.mdx:48]:
```suggestion
Consul Kubernetes is [certified](https://marketplace.cloud.vmware.com/services/details/hashicorp-consul-1?slug=true) for both VMware Tanzu Kubernetes Grid, and VMware Tanzu Kubernetes Integrated Edition.
```

**Review by karl-cardenas-coding — APPROVED** [review]

**Review by bear359 — APPROVED** [review]

**Comment by david-yu** [comment]
Added section around platform specific compatibility to group Openshift and VMware Tanzu, and fixing formatting issue for shell-session in Systems Integration of Vault Secrets backend.

**Comment by david-yu** [comment]
Will merge with Consul K8s release tomorrow.

**Comment by karl-cardenas-coding** [comment]
@david-yu If you remove cherry-pick and merge, then this will become available when 1.12 is released.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/648040.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 5016ccb51ee0f1bfdebf1a3e35cfa5914315c278 onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 5016ccb51ee0f1bfdebf1a3e35cfa5914315c278 onto `release/1.12.x` succeeded!

**Pull request merged**